tarogarden/planter.go

```diff
@@ -486,6 +490,19 @@ func (c *ChainPlanter) prepTaroSeedling(ctx context.Context,
 		defer cancel()
 		err = c.cfg.Log.CommitMintingBatch(ctx, newBatch)
 		if err != nil {
+			// Handle SQLite asset seedling name unique constraint
+			// violation error.
+			var sqlError *sqlite.Error
+			if errors.As(err, &sqlError) &&
+				sqlError.Code() == sqlite3.SQLITE_CONSTRAINT_UNIQUE &&
+				strings.Contains(err.Error(),
+					"asset_seedlings.asset_name") {
+
+				return false, &ErrDuplicateSeedlingName{
+					Name: req.AssetName,
+				}
+			}
```

Comment:
@Roasbeef I think this is what you mean when you said that I should check the error outside the closure. Because we're checking at a higher level, I thought I'd include:

```go
				strings.Contains(err.Error(),
					"asset_seedlings.asset_name")
```

because there might be other unique constraints (in the future) at play during batching and so on.

I used `sqlite3.SQLITE_CONSTRAINT_UNIQUE` instead of `SQLITE_CONSTRAINT_CHECK` because that's the sqlite error code returned on my machine. The test passes on my machine, I wander if it will in CI and on your machine.

Reply:
Cool confirmed that this passes on my machine.